src/realm-settings/NewAttributeSettings.tsx

```diff
@@ -1,23 +1,205 @@
-        className="pf-u-mt-lg"
+    <UserProfileProvider>
+      <ScrollForm
+        sections={[
+          t("generalSettings"),
+          t("permission"),
+          t("validations"),
+          t("annotations"),
+        ]}
       >
-        <p>{t("createAttribute")}</p>
-      </FormAccess>
-    </PageSection>
+        <AttributeGeneralSettings form={form} />
+        <AttributePermission form={form} />
+        <AttributeValidations />
+        <AttributeAnnotations form={form} />
+      </ScrollForm>
+      <Form onSubmit={form.handleSubmit(save)}>
+        <ActionGroup className="keycloak__form_actions">
+          <Button
+            variant="primary"
+            type="submit"
+            data-testid="attribute-create"
+          >
+            {t("common:create")}
+          </Button>
+          <Button
+            variant="link"
+            onClick={() =>
+              history.push(toUserProfile({ realm, tab: "attributes" }))
+            }
+            data-testid="attribute-cancel"
+          >
+            {t("common:cancel")}
+          </Button>
+        </ActionGroup>
+      </Form>
+    </UserProfileProvider>
+  );
+};
+
+export default function NewAttributeSettings() {
+  const { realm: realmName } = useRealm();
+  const adminClient = useAdminClient();
+  const form = useForm<UserProfileConfig>();
+  const { t } = useTranslation("realm-settings");
+  const history = useHistory();
+  const { addAlert, addError } = useAlerts();
+  const [config, setConfig] = useState<UserProfileConfig | null>(null);
+  const [clientScopes, setClientScopes] =
+    useState<ClientScopeRepresentation[]>();
+
+  useFetch(
+    () => adminClient.users.getProfile({ realm: realmName }),
+    (config) => setConfig(config),
+    []
+  );
+
+  useFetch(
+    () => adminClient.clientScopes.find(),
+    (clientScopes) => {
+      setClientScopes(clientScopes);
+    },
+    []
+  );
+
+  const save = async (profileConfig: UserProfileAttributeType) => {
+    const scopeNames = clientScopes?.map((clientScope) => clientScope.name);
+
+    const selector = {
+      scopes: profileConfig.enabledWhen
+        ? scopeNames
+        : profileConfig.selector?.scopes,
+    };
+
+    const required = {
+      roles: profileConfig.roles,
+      scopes: profileConfig.requiredWhen
+        ? scopeNames
+        : profileConfig.scopeRequired,
+    };
+
+    const validations = profileConfig.validations;
+
+    const permissions = {
+      view: profileConfig.view,
+      edit: profileConfig.edit,
+    };
+
+    const annotations = (profileConfig.annotations! as KeyValueType[]).reduce(
+      (obj, item) => Object.assign(obj, { [item.key]: item.value }),
+      {}
+    );
+
+    const newAttribute = [
+      {
+        name: profileConfig.name,
+        displayName: profileConfig.displayName,
+        required: required,
+        validations: validations,
+        selector: selector,
+        permissions: permissions,
+        annotations: annotations,
```

Comment:
Can use the short hand like this:
```suggestion
        required,
        validations,
        selector,
        permissions,
        annotations,
```